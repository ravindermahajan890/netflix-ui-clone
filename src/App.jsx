import { useState } from 'react';
import './App.css';
import { Button } from '@mui/material';

function App() {
  const [count, setCount] = useState(0);

  return (
    <div className="App">
      <h1>TAMASHFLIX</h1>
      <div className="card">
        <button onClick={() => setCount((count) => count + 1)}>
          Count is {count}
        </button>
      </div>
      <Button variant="contained" color="success" onClick={() => setCount(0)}>
        Reset the count
      </Button>
      <p className="read-the-docs">Basic homepage for TAMASHFLIX</p>
    </div>
  );
}

export default App;
