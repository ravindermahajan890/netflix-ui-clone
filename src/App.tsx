import { useState } from 'react';
import { Button } from '@mui/material';
import './App.css';

const App = () => {
  const [count, setCount] = useState(0);

  return (
    <div className="App">
      <h1>TAMASHFLIX</h1>
      <div className="card">
        <button type="button" onClick={() => setCount((prevCount) => prevCount + 1)}>
          Count is {count}
        </button>
      </div>
      <Button variant="contained" color="success" onClick={() => setCount(0)}>
        Reset the count
      </Button>
      <p className="read-the-docs">Basic homepage for TAMASHFLIX</p>
    </div>
  );
};

export default App;
