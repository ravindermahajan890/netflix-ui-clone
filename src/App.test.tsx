import { fireEvent, render } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('should render', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should increase count', () => {
    const { getByText } = render(<App />);

    fireEvent.click(getByText('Count is 0'));

    expect(getByText('Count is 1')).toBeInTheDocument();
  });

  it('should reset the count', () => {
    const { getByText } = render(<App />);

    fireEvent.click(getByText('Count is 0'));
    fireEvent.click(getByText('Reset the count'));

    expect(getByText('Count is 0')).toBeInTheDocument();
  });
});
