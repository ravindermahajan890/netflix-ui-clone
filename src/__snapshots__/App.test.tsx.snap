// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      TAMASHFLIX
    </h1>
    <div
      class="card"
    >
      <button
        type="button"
      >
        Count is 0
      </button>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-ke5b6m-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset the count
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <p
      class="read-the-docs"
    >
      Basic homepage for TAMASHFLIX
    </p>
  </div>
</DocumentFragment>
`;
